tiny_dnn/layers/batch_normalization_layer.h

```diff
@@ -91,15 +91,15 @@ class batch_normalization_layer : public layer {
     tensor_t &prev_delta      = *in_grad[0];
     tensor_t &curr_delta      = *out_grad[0];
     const tensor_t &curr_out  = *out_data[0];
-    serial_size_t num_samples = static_cast<serial_size_t>(curr_out.size());
+    size_t num_samples = static_cast<size_t>(curr_out.size());
```

Comment:
`static_cast<size_t>` isn't needed since you changed the type of the local variable to `size_t`.